Lib/glyphsLib/builder/builders.py

```diff
@@ -247,6 +264,105 @@ def instance_data(self):
+                    "{bracket_axis_minimum}, maximum {bracket_axis_maximum}.".format(
+                        glyph_name=layer.parent.name,
+                        layer_name=layer.name,
+                        bracket_axis_name=bracket_axis.name,
+                        bracket_axis_minimum=bracket_axis_min,
+                        bracket_axis_maximum=bracket_axis_max,
+                    )
+                )
+            bracket_layer_map[bracket_crossover].append(layer)
+
+        # Map crossovers to glyph names, i.e. if glyph "x" and "y" have the bracket
+        # layers "[300]" and "[600]", crossovers will be
+        # {"x": [300, 600], "y": [300, 600]}. This helps with defining overlaps in the
+        # replacment rules below.
+        # Note: sort by location to ensure value list sortedness for roundtrip
+        # stability on Python 2.
+        crossovers = defaultdict(list)  # type: Dict[str, List[int]]
+        for location, layers in sorted(bracket_layer_map.items()):
+            glyph_name_set = {l.parent.name for l in layers}
+            for glyph_name in glyph_name_set:
+                crossovers[glyph_name].append(location)
+
+        # Copy bracket layers to their own glyphs.
+        for location, layers in bracket_layer_map.items():
+            for layer in layers:
+                ufo_font = self._sources[
+                    layer.associatedMasterId or layer.layerId
+                ].font.layers.defaultLayer
+                ufo_glyph_name = "{layer_parent_name}.BRACKET.{location}".format(
+                    layer_parent_name=layer.parent.name, location=location
+                )
+                ufo_glyph = ufo_font.newGlyph(ufo_glyph_name)
+                self.to_ufo_glyph(ufo_glyph, layer, layer.parent)
+                ufo_glyph.unicodes = []
+                ufo_glyph.lib[GLYPHLIB_PREFIX + "_originalLayerName"] = layer.name
+
+        def pairwise(iterable):
+            "s -> (s0,s1), (s1,s2), (s2, s3), ..."
+            a, b = itertools.tee(iterable)
+            next(b, None)
+            return zip(a, b)
+
+        # Generate rules for the bracket layers.
+        for glyph_name, axis_crossovers in crossovers.items():
+            for crossover_min, crossover_max in pairwise(
+                axis_crossovers + [bracket_axis_max]
+            ):
+                rule = designspaceLib.RuleDescriptor()
+                rule.name = "{glyph_name}.BRACKET.{crossover_min}".format(
+                    glyph_name=glyph_name, crossover_min=crossover_min
+                )
+                rule.conditionSets.append(
+                    [
+                        {
+                            "name": bracket_axis.name,
+                            "minimum": crossover_min,
+                            "maximum": crossover_max,
+                        }
+                    ]
+                )
+                rule.subs.append((glyph_name, rule.name))
```

Comment:
instead of rule.name here write ufo_glyph_name so that if someone wants to change the name of the rule then it won't break the substitution